framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -330,26 +342,39 @@ export class CommitPool {
 				...this._nonGossipedCommitsLocal.getByHeight(nextHeight),
 				...this._gossipedCommits.getByHeight(nextHeight),
 			];
-			const nextValidators = singleCommits.map(commit => commit.validatorAddress);
 			let aggregateBFTWeight = BigInt(0);
 
 			// Assume BFT parameters exist for next height
 			const { validators: bftParamValidators, certificateThreshold } =
 				await this._bftMethod.getBFTParameters(methodContext, nextHeight);
 
+			const filteredSingleCommits = singleCommits.filter(commit => {
+				const foundValidator = bftParamValidators.find(v =>
+					v.address.equals(commit.validatorAddress),
+				);
+				if (foundValidator && foundValidator.bftWeight === BigInt(0)) {
+					return false;
+				}
+				return true;
+			});
+			const nextValidators = filteredSingleCommits.map(commit => commit.validatorAddress);
 			for (const matchingAddress of nextValidators) {
 				const bftParamsValidatorInfo = bftParamValidators.find(bftParamValidator =>
 					bftParamValidator.address.equals(matchingAddress),
 				);
 				if (!bftParamsValidatorInfo) {
 					throw new Error('Validator address not found in commit pool');
 				}
+				// Skip validators with BFT Weight zero when someone is running node with standby validators
+				if (bftParamsValidatorInfo.bftWeight === BigInt(0)) {
```

Comment:
This should never happen, since the bft weights are filtered out, right?

Reply:
currently it fails at `Validator address not found in commit pool'` if we use `getBFTParametersActiveValidators` and if we use `getBFTParameters` then we need to make sure to ignore the standby delegates